twake/backend/node/src/services/preview/src/previews/services/index.ts

```diff
@@ -0,0 +1,42 @@
+import { PreviewServiceAPI } from "../api";
+import { generatePreview as thumbnailsFromImages } from "./image";
+import { convertFromOffice } from "./office";
+import { convertFromPdf } from "./pdf";
+import { isFileType } from "./mime";
+import { pdfExtensions, officeExtensions, imageExtensions } from "./mime";
+
+export function getService(): PreviewServiceAPI {
+  return getServiceInstance();
+}
+
+function getServiceInstance(): PreviewServiceAPI {
+  return new Service();
+}
+class Service implements PreviewServiceAPI {
+  version: "1";
+
+  constructor() {}
+
+  async generateThumbnails(
+    mime: string,
+    inputPath: string,
+    outputPath: string,
+    outputExtension: string,
+    numberOfPages: number,
+  ): Promise<any> {
+    if (isFileType(mime, inputPath.split("/").pop(), officeExtensions)) {
+      const pdfPath = await convertFromOffice(inputPath, outputPath, numberOfPages);
+      const thumbnailPath = await convertFromPdf(pdfPath, numberOfPages);
+      thumbnailsFromImages(thumbnailPath, outputPath, outputExtension);
```

Comment:
you must `return` here. If not you are moving to the next if condition which is useless